src/Streamly/Internal/Data/Unfold/Type.hs

```diff
@@ -144,11 +148,13 @@ apDiscardSnd (Unfold _step1 _inject1) (Unfold _step2 _inject2) = undefined
 data Cross a s1 b s2 = CrossOuter a s1 | CrossInner a s1 b s2
 
 -- | Create a cross product (vector product or cartesian product) of the
--- output streams of two unfolds.
+-- output streams of two unfolds using a monadic combining function.
 --
-{-# INLINE_NORMAL cross #-}
-cross :: Monad m => Unfold m a b -> Unfold m a c -> Unfold m a (b, c)
-cross (Unfold step1 inject1) (Unfold step2 inject2) = Unfold step inject
+-- /Pre-release/
+{-# INLINE_NORMAL crossWithM #-}
+crossWithM :: Monad m =>
```

Comment:
Can you also benchmark this? Or you can open a related issue and assign it to me.

Reply:
`apply` is benchmarked through `toNullAp` benchmark. apply indirectly benchmarks this.